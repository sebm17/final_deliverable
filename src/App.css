.b1Window {
  background: rgb(183, 183, 209);
  width: 100%;
  margin-left: 20px;
  margin-right: 20px;
}

.inputList {
  max-height: 300px;
  overflow: auto;
}

.b1Card:hover {
  background-color: #b8c8f7;
  transition: 0.7s;
}

.selectedCard {
  background-color: rgb(97, 97, 228);
}

.downloadButton {
  width: 100%;
  height: 100%;
}

input.error {
  background-color: red !important;
}

input:focus {
  background-color: lightblue !important;
}
