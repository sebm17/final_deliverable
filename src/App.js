import React, { createContext, useReducer } from "react";
import "./App.css";
import { Header, AddB1, B1Card, B1List, B1Window } from "./components";
import { B1FileInputProvider } from "./B1Context";

const initialB1Files = [
  {
    id: "1",
    mpan: "1591029068761",
    mprn: "7471856205",
    jad: "jad1",
    jtc: "jtc1"
  },
  {
    id: "2",
    mpan: "mpan2",
    mprn: "mprn2",
    jad: "jad2",
    jtc: "jtc2"
  }
];

function b1Reducer(state, action) {
  const { selectedInputId, b1Inputs } = state;
  switch (action.type) {
    case "ADD_B1_FILE":
      // ID is whatever the last one is + 1
      const newId = b1Inputs[b1Inputs.length - 1]
        ? parseInt(b1Inputs[b1Inputs.length - 1].id, 10) + 1
        : 1;
      return {
        selectedInputId: newId,
        b1Inputs: [...b1Inputs, { ...action.payload, id: newId }]
      };
    case "REMOVE_B1_FILE":
      const newInputs = b1Inputs.filter((b1File) => b1File.id !== action.id);
      const newSelectedInputId =
        action.id === selectedInputId ? undefined : selectedInputId;
      return { selectedInputId: newSelectedInputId, b1Inputs: newInputs };
    case "SET_CURRENT_B1_FILE":
      return { ...state, selectedInputId: action.id };
    default:
      console.warn("unknown action received type=", action.type);
      return state;
  }
}

function App() {
  const [state, dispatch] = useReducer(b1Reducer, { b1Inputs: initialB1Files });

  return (
    <B1FileInputProvider value={[state, dispatch]}>
      <div className="ui vertically divided grid">
        <div className="one column row">
          <Header />
        </div>
        <div className="two column row">
          <div className="column">
            <AddB1 />
          </div>
          <div className="column">
            <B1List b1Files={state.b1Inputs} />
          </div>
        </div>
        <B1Window
          selectedInput={state.b1Inputs.find(
            (input) => input.id === state.selectedInputId
          )}
        />
      </div>
    </B1FileInputProvider>
  );
}

export default App;
