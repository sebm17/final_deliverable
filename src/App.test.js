import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

test("App renders as expected", () => {
  const wrapper = render(<App />);
  expect(wrapper).toMatchSnapshot();
});
