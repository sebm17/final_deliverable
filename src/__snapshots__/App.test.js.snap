// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui vertically divided grid"
      >
        <div
          class="one column row"
        >
          <div
            class="ui fixed menu"
          >
            <div
              class="ui container center"
            >
              <h1>
                B1 File Creator
              </h1>
            </div>
          </div>
        </div>
        <div
          class="two column row"
        >
          <div
            class="column"
          >
            <div
              class="ui main"
            >
              <h1
                class="title"
              >
                Input B1 data
              </h1>
              <form
                class="ui form"
              >
                <div
                  class="field"
                >
                  <label>
                    MPAN
                  </label>
                  <input
                    data-testid="mpan"
                    name="mpan"
                    placeholder="mpan"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="field"
                >
                  <label>
                    MPRN
                  </label>
                  <input
                    data-testid="mprn"
                    name="mprn"
                    placeholder="mprn"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="field"
                >
                  <label>
                    Job Appointment Date
                  </label>
                  <input
                    data-testid="jad"
                    name="jad"
                    placeholder="jad"
                    type="date"
                    value=""
                  />
                </div>
                <div
                  class="field"
                >
                  <label>
                    Job Type Code
                  </label>
                  <input
                    data-testid="jtc"
                    name="jtc"
                    placeholder="jtc"
                    type="text"
                    value=""
                  />
                </div>
              </form>
              <button
                class="ui button blue"
                disabled=""
              >
                Process B1 file
              </button>
              <button
                class="ui button grey"
              >
                Reset
              </button>
            </div>
          </div>
          <div
            class="column"
          >
            <h1>
              Previous files
            </h1>
            <div
              class="ui celled list inputList"
            >
              <div
                class="item b1Card"
                data-testid="card"
              >
                <div
                  class="content"
                >
                  <div
                    class="header"
                  >
                    1
                  </div>
                  <div>
                    <b>
                      mprn
                    </b>
                    : 
                    7471856205
                  </div>
                  <div>
                    <b>
                      mpan
                    </b>
                    : 
                    1591029068761
                  </div>
                  <div>
                    <b>
                      job appointment date
                    </b>
                    : 
                    jad1
                  </div>
                  <div>
                    <b>
                      job type code
                    </b>
                    : 
                    jtc1
                  </div>
                </div>
                <i
                  class="trash alternate outline icon"
                  data-testid="bin"
                  style="color: red; margin-top: 7px;"
                />
              </div>
              <div
                class="item b1Card"
                data-testid="card"
              >
                <div
                  class="content"
                >
                  <div
                    class="header"
                  >
                    2
                  </div>
                  <div>
                    <b>
                      mprn
                    </b>
                    : 
                    mprn2
                  </div>
                  <div>
                    <b>
                      mpan
                    </b>
                    : 
                    mpan2
                  </div>
                  <div>
                    <b>
                      job appointment date
                    </b>
                    : 
                    jad2
                  </div>
                  <div>
                    <b>
                      job type code
                    </b>
                    : 
                    jtc2
                  </div>
                </div>
                <i
                  class="trash alternate outline icon"
                  data-testid="bin"
                  style="color: red; margin-top: 7px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui vertically divided grid"
    >
      <div
        class="one column row"
      >
        <div
          class="ui fixed menu"
        >
          <div
            class="ui container center"
          >
            <h1>
              B1 File Creator
            </h1>
          </div>
        </div>
      </div>
      <div
        class="two column row"
      >
        <div
          class="column"
        >
          <div
            class="ui main"
          >
            <h1
              class="title"
            >
              Input B1 data
            </h1>
            <form
              class="ui form"
            >
              <div
                class="field"
              >
                <label>
                  MPAN
                </label>
                <input
                  data-testid="mpan"
                  name="mpan"
                  placeholder="mpan"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="field"
              >
                <label>
                  MPRN
                </label>
                <input
                  data-testid="mprn"
                  name="mprn"
                  placeholder="mprn"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="field"
              >
                <label>
                  Job Appointment Date
                </label>
                <input
                  data-testid="jad"
                  name="jad"
                  placeholder="jad"
                  type="date"
                  value=""
                />
              </div>
              <div
                class="field"
              >
                <label>
                  Job Type Code
                </label>
                <input
                  data-testid="jtc"
                  name="jtc"
                  placeholder="jtc"
                  type="text"
                  value=""
                />
              </div>
            </form>
            <button
              class="ui button blue"
              disabled=""
            >
              Process B1 file
            </button>
            <button
              class="ui button grey"
            >
              Reset
            </button>
          </div>
        </div>
        <div
          class="column"
        >
          <h1>
            Previous files
          </h1>
          <div
            class="ui celled list inputList"
          >
            <div
              class="item b1Card"
              data-testid="card"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                >
                  1
                </div>
                <div>
                  <b>
                    mprn
                  </b>
                  : 
                  7471856205
                </div>
                <div>
                  <b>
                    mpan
                  </b>
                  : 
                  1591029068761
                </div>
                <div>
                  <b>
                    job appointment date
                  </b>
                  : 
                  jad1
                </div>
                <div>
                  <b>
                    job type code
                  </b>
                  : 
                  jtc1
                </div>
              </div>
              <i
                class="trash alternate outline icon"
                data-testid="bin"
                style="color: red; margin-top: 7px;"
              />
            </div>
            <div
              class="item b1Card"
              data-testid="card"
            >
              <div
                class="content"
              >
                <div
                  class="header"
                >
                  2
                </div>
                <div>
                  <b>
                    mprn
                  </b>
                  : 
                  mprn2
                </div>
                <div>
                  <b>
                    mpan
                  </b>
                  : 
                  mpan2
                </div>
                <div>
                  <b>
                    job appointment date
                  </b>
                  : 
                  jad2
                </div>
                <div>
                  <b>
                    job type code
                  </b>
                  : 
                  jtc2
                </div>
              </div>
              <i
                class="trash alternate outline icon"
                data-testid="bin"
                style="color: red; margin-top: 7px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
